hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaControllerManagement.java

```diff
@@ -375,6 +375,14 @@ public boolean hasTargetArtifactAssigned(final long targetId, final String sha1H
         return actionRepository.findByExternalRefInAndActive(externalRefs, true);
     }
 
+    @Override
+    public void deleteExistingTarget(@NotEmpty String controllerId) {
+        final JpaTarget target = (JpaTarget) targetRepository.findByControllerId(controllerId)
+                .orElseThrow(() -> new EntityNotFoundException(Target.class, controllerId));
```

Comment:
does this Exception needs to be thrown?

Reply:
It should just be a warning. 
Check also in other codes snippets were targets are deleted

Reply:
The exception throwing is necessary to forward invalid incoming messages to the deadletter queue. 
However, exception catching should be handled dmf globally to avoid logging the whole stacktrace of every thrown exception inside JpaControllerManagement.

Reply:
...to be clarified if deadletter forwarding is even necessary here.